### PR TITLE
feat: add amendments enabled on each network

### PR DESCRIPTION
**Pull request by pdp2121**
## High Level Overview of Change

Create a table with the following information to track amendments enabled on each network:

- AmendmentID

- NetworkID

- ledger_index

- tx_hash

- date

Also add a function to manually update the table using a JSON file (runs whenever re-deployed).

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Test Plan

Works on staging.

### CODE REVIEW

File: src/connection-manager/connections.ts [hunk 1]
```diff
@@ -37,6 +45,21 @@ function subscribe(ws: WebSocket): void {
   )
 }
 
+/**
+ * Sends a ledger_entry WebSocket request to retrieve amendments enabled on a network.
+ *
+ * @param ws - A WebSocket object.
+ */
+function ledger_entry(ws: WebSocket): void {
```
Comment (mvadari) [inline src/connection-manager/connections.ts:53]: This shouldn't be called just `ledger_entry`, it should be something like `getAmendmentLedgerEntry` - it's not just a wrapper for _any_ `ledger_entry` request.

File: src/shared/types/index.ts [hunk 2]
```diff
@@ -29,6 +29,37 @@ interface StreamLedger {
...
+}
+
+interface LedgerEntryMajority {
+  Majority: {
+    Amendment: string
+    CloseTime: number
+  }
+}
+
+interface LedgerEntryAmendments {
```
Comment (mvadari) [inline src/shared/types/index.ts:44]: nit:
```suggestion
interface LedgerEntryAmendmentsResponse {
```

File: src/connection-manager/connections.ts [hunk 3]
```diff
@@ -89,6 +123,9 @@ async function setHandlers(
       void saveNodeWsUrl(ws.url, true)
       connections.set(ip, ws)
       subscribe(ws)
+      if (entry) {
```
[the remark this answers is not in the document]
Comment (pdp2121) [reply src/connection-manager/connections.ts:126]: We just need to call `ledger_entry` once per network to get currently enabled amendments list so I just call it using entry nodes

File: src/connection-manager/connections.ts [hunk 4]
```diff
@@ -6,13 +8,18 @@ import {
   clearConnectionsDb,
   getNetworks,
   saveAmendmentsEnabled,
+  saveAmendmentEnabled,
```
Comment (mvadari) [inline src/connection-manager/connections.ts]: Why are these two separate functions?

**Review by mvadari — COMMENTED** [review]
For ease of reviewing, please set up the PRs so they merge one into another. You have write access on the repo so you can create + push branches directly off of the main repo instead of off of a fork.

**Pull request closed without merging**